### ISSUE TITLE
[BUG] Prevent props from passing through to DOM elements

### ISSUE THREAD

**Issue by Simba14** [framer/motion#1406]
It seems the issue detailed in [Issue 272](https://github.com/framer/motion/issues/272) has re-occurred when using the latest version `5.5.5` in combination with styled components.  I have tried earlier versions but same error is thrown: 
`React does not recognize the `customProp` prop on a DOM element.`

It seems all props are passed down by motion to the DOM element. They should be stripped out to prevent this error.
I have seen that an earlier fix was to utilise ` @emotion/is-prop-valid` as an optional dependency.
Perhaps this dependency is not correctly being used in cases when it should?

**3. IMPORTANT: Provide a CodeSandbox reproduction of the bug**
https://github.com/mattbrannon/motion-console-warning
Unfortunately unable to reproduce via sandbox, but @mattbrannon made this public repo specifically to demonstrate the issue. 

**4. Steps to reproduce**
`customProp` is passed on to the DOM element when using Styled Components
```
const Custom = styled(motion.div)`
  opacity: ${({ customProp }) => customProp};
`

...
return (
  <Custom customProp={1}></CustomProp>
)
```
**5. Expected behavior**

For custom props to filtered out and not passed on to the DOM element created by motion.

**Comment by GR34SE** [comment]
I have encountered the same issue right after upgrading from v3.10.6 to the latest v5.5.5. It seems like it appeared since v5.0.0 as it works well in 4.1.17

**Comment by Simba14** [comment]
A workaround is using Styled Components [transient props](https://styled-components.com/docs/api#transient-props) for anyone blocked by this

**Comment by mattgperry** [comment]
I wonder if the move to `.mjs` es files somehow breaks the ability to use optional dependencies. I'll take a look

**Comment by mattgperry** [comment]
Yeah, in that sandbox by moving the optional import outside the `try` I get `require is not defined`

**Comment by nataliarossini** [comment]
I had the same issue and fixed by only adding a $ to my props like so `$customProp`
Apparently this release solves the issue with props filtering
https://github.com/styled-components/styled-components/releases/tag/v5.1.0

**Comment by mattgperry** [comment]
I've just published `5.6.0` which will let you pass a custom function:

```jsx
import { MotionConfig } from "framer-motion"
import isValidProp from "@emotion/is-prop-valid"

function App() {
  return (
    <MotionConfig isValidProp={isValidProp}>

   </MotionConfig>
  )
}
```
